lib/srv/discovery/discovery_test.go

```diff
@@ -1302,6 +1300,97 @@ func TestDiscoveryKubeServices(t *testing.T) {
 	}
 }
 
+// TestDiscoveryKubeServicesInterval tests that the poll interval is honored for Kube App Discovery.
+func TestDiscoveryKubeServicesInterval(t *testing.T) {
+	const mainDiscoveryGroup = "main"
+	clock := clockwork.NewFakeClock()
+	waitForReconcileTimeout := 5 * time.Second
+	tickDuration := 200 * time.Millisecond
+	ctx := context.Background()
```

Comment:
Suggestion: use the test aware context provided by testing.T. See https://pkg.go.dev/testing#T.Context.
```suggestion
	ctx := t.Context()
```